pymeasure/instruments/keysight/keysight34980A.py

```diff
@@ -0,0 +1,628 @@
+        If you omit the optional <ch_list> parameter, this command applies to the internal DMM, 
+        independent of any channels or a scan list. 
+        """
+        if (chan == ""):
+            self.write(":UNIT:TEMPerature?") 
+            temp_values = self.read()
+            return(str(temp_values))
+        else:
+            self.write(":UNIT:TEMPerature? (@" + str(chan) + ")") 
+            temp_values = self.read()
+            return(str(temp_values))
+
+    def format_reading_alarm_enable(self):
+        """
+        This command enables the inclusion of alarm information with data retrieved 
+        by the READ? command, the FETCh? command, or other queries of scan results.
+        """
+        self.write(":FORMat:READing:ALARm ON")
+        while((self.values(":FORMat:READing:ALARm?")) == 0):
+            self.write(":FORMat:READing:ALARm ON")
+
+    def format_reading_alarm_disable(self):
+        """
+        This command disables the inclusion of alarm information with data retrieved 
+        by the READ? command, the FETCh? command, or other queries of scan results.
+        """
+        self.write(":FORMat:READing:ALARm OFF")
+        while((self.values(":FORMat:READing:ALARm?")) == 1):
+            self.write(":FORMat:READing:ALARm OFF")
+
+    def format_reading_channel_enable(self):
+        """
+        This command enables the inclusion of channel number information with data retrieved 
+        by the READ? command, the FETCh? command, or other queries of scan results.
+        """
+        self.write(":FORMat:READing:CHANnel ON")
+        while((self.values(":FORMat:READing:CHANnel?")) == 0):
+            self.write(":FORMat:READing:CHANnel ON")
+
+    def format_reading_channel_disable(self):
+        """
+        This command disables the inclusion of channel number information with data retrieved 
+        by the READ? command, the FETCh? command, or other queries of scan results.
+        """
+        self.write(":FORMat:READing:CHANnel OFF")
+        while((self.values(":FORMat:READing:CHANnel?")) == 1):
+            self.write(":FORMat:READing:CHANnel OFF")
+
+    def format_reading_time_enable(self):
+        """
+        This command enables the inclusion of time stamp with data retrieved 
+        by the READ? command, the FETCh? command, or other queries of scan results.
+        """
+        self.write(":FORMat:READing:TIME ON")
+        while((self.values(":FORMat:READing:TIME?")) == 0):
+            self.write(":FORMat:READing:TIME ON")        
+
+    def format_reading_time_disable(self):
+        """
+        This command disables the inclusion of time stamp with data retrieved 
+        by the READ? command, the FETCh? command, or other queries of scan results.
+        """
+        self.write(":FORMat:READing:TIME OFF")
+        while((self.values(":FORMat:READing:TIME?")) == 1):
+            self.write(":FORMat:READing:TIME OFF") 
```

Comment:
I think these (or at least most of these) can be nicely converted into a properties (also holds for a number of other methods, but the ones over here seem like a good starting point): for example `Instrument.measurement` for the measure_xx methods and `Instrument.control` for set_temperature_unit and is_temperature_unit.